### ISSUE TITLE
`MissingMethodException` in `ComponentTask.Internal.TraceLogger.Initialize()`

### ISSUE THREAD

**Issue by eirikwah** [BastianBlokland/componenttask-unity#20]
Using Unity 2019.3.0b11, and ComponentTask-Unity version 1.8. Built as Windows Standalone in Unity. When starting the application, I get this error in the log:

```
MissingMethodException: Default constructor not found for type System.Diagnostics.SystemDiagnosticsSection
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic) [0x0007b] in <6c049d01a5ff46bfa433c40b27bfb01e>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00009] in <6c049d01a5ff46bfa433c40b27bfb01e>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00027] in <6c049d01a5ff46bfa433c40b27bfb01e>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00020] in <6c049d01a5ff46bfa433c40b27bfb01e>:0 
  at System.Configuration.ConfigInfo.CreateInstance () [0x00026] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.SectionInfo.CreateInstance () [0x00000] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.Configuration.GetSectionInstance (System.Configuration.SectionInfo config, System.Boolean createDefaultInstance) [0x0001c] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.ConfigurationSectionCollection.get_Item (System.String name) [0x0002c] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.Configuration.GetSection (System.String sectionName) [0x0001e] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.ClientConfigurationSystem.System.Configuration.Internal.IInternalConfigSystem.GetSection (System.String configKey) [0x00006] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.ConfigurationManager.GetSection (System.String sectionName) [0x00005] in <eea339da6b5e4d4bb255bfef95601890>:0 
  at System.Configuration.PrivilegedConfigurationManager.GetSection (System.String sectionName) [0x00000] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at System.Diagnostics.DiagnosticsConfiguration.GetConfigSection () [0x00000] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at System.Diagnostics.DiagnosticsConfiguration.Initialize () [0x0002a] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at System.Diagnostics.DiagnosticsConfiguration.get_IndentSize () [0x00000] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at System.Diagnostics.TraceInternal.InitializeSettings () [0x0004e] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at System.Diagnostics.TraceInternal.get_Listeners () [0x00000] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at System.Diagnostics.Trace.get_Listeners () [0x00000] in <43cff77c8e644fb3bd45df5f20310d13>:0 
  at ComponentTask.Internal.TraceLogger.Initialize () [0x0000c] in D:\a\1\s\Library\PackageCache\com.bastianblokland.componenttask@e2759e9a8808128e548ecdf7114d02f144657034\src\ComponentTask\Internal\TraceLogger.cs:39 
```

I am guessing this can be fixed by including a link.xml file, as described here: https://docs.unity3d.com/2019.3/Documentation/Manual/IL2CPP-BytecodeStripping.html

I will try to fork and make a pull request to fix this (when I get the time).

**Comment by BastianBlokland** [comment]
Thanks for reporting, couldn't reproduce on latest non beta (`2019.2.13f1`). Will make a pr with the upgrade to `2019.2.13f1` anyway so we're up to date again.

I'll see if i'm able to test with `2019.3.0b11` somewhere today.

**Comment by BastianBlokland** [comment]
Well update to `2019.2.13f1` is also a major PITA, not because of actual code but getting the ci pipeline to work with it. Apparently allot has changed in way modules work. I will continue this a bit later.

**Comment by BastianBlokland** [comment]
@eirikwah I cannot reproduce it on `2019.3.0b11` with il2cpp windows standalone. Even with the managed stripping set to `high`.
![ProjectSettings](https://user-images.githubusercontent.com/14230060/69585041-4b18cd00-0fe7-11ea-9c9b-c8d5a6718fc0.PNG)
I must be using slightly different settings, what project settings are you building with?

**Comment by eirikwah** [comment]
This is my build settings:
![image](https://user-images.githubusercontent.com/1539783/69627381-c6f83100-104a-11ea-900b-262dc2036ba5.png)

![image](https://user-images.githubusercontent.com/1539783/69627537-13dc0780-104b-11ea-8a6c-9e2207900f6a.png)

...and now I am realizing that I am building with Mono, and not il2cpp as I thought. I will investiagate some more. Sorry for the confusion.

Also maybe relevant: I have included moq.dll (v4.13.1) and its dependencies Castle.Core.dll (v4.4.0), System.Runtime.CompilerServices.Unsafe.dll (v4.5.0), and System.Threading.Tasks.Extensions.dll (v4.5.1). And we are using Zenject (Extenject) version 9.1.9.

**Comment by BastianBlokland** [comment]
Thanks for looking more at this.

That it happens on the mono runtime is even stranger, as it should be able to jit compile it then 😕. In any case this project should't use anything that a aot compiler (or stripper) wouldn't be able to figure out (no reflection or any form of dynamic code gen). 

Will do some more investigation this evening.

**Comment by BastianBlokland** [comment]
I am able to reproduce with the `mono` backend and and `4x` api compat (`netstandard 2.0` works fine for me). I'll experiment a bit if there is a workaround with a `link.xml`.

**Comment by BastianBlokland** [comment]
Ended up disabling the TraceLogger for `net 4.6` in combination with the `mono` backend. Not worth the hassle as its only a log sink for`Debug.Assert` and `Debug.Fail` and either triggering would mean a serious bug in this repository. Is in included in release `v1.8.2`.

**Comment by eirikwah** [comment]
Thank you! Your fix works nicely.
